core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -5682,11 +5682,17 @@ public static List mapEntries(Map<Object, Object> map) {
 
   /** Support the MAP_KEYS function. */
   public static List mapKeys(Map map) {
+    if (map.keySet().contains(null)) {
```

Comment:
what is interesting is that you detect that a map key is null only when you call the function map_keys. But shouldn't this be detected when inserting something into the map, if it is indeed illegal?

If it's not illegal to insert entries with null keys, maybe the error message should say that the function map_keys does not support null keys in maps.

Reply:
I don't know what parameters you are talking about.
I am saying that the error message does not seem right. The error is not inserting an entry with a null key in a map, it's using a map with a null key in the function mapKeys

Reply:
Thank you for your patience, I misunderstood you, I will fix it.